Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/MockJSONHttpClient.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Created by Dirk Jäckel on 08.07.22.
+// Copyright (c) 2022 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Multisig
+
+class MockJSONHttpClient {
+
+    typealias TxData = SCGModels.TxData
+
+    var decoder: JSONDecoder = JSONDecoder()
+
+    init() {
+        decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .millisecondsSince1970
+    }
+
+    private func loadAndParseFile(fileName: String) throws -> SCGModels.TransactionDetails? {
+        let data = jsonData(fileName)
+        return try decoder.decode(SCGModels.TransactionDetails.self, from: data)
+    }
+
+    private func jsonData(_ name: String) -> Data {
+        try! Data(contentsOf: Bundle(for: Self.self).url(forResource: name, withExtension: "json")!)
+    }
+
+    func asyncExecute<T: JSONRequest>(request: T, completion: @escaping (Result<T.ResponseType, Error>) -> ()) -> URLSessionTask? {
+        //Check if we can handle the request
+        if request is TransactionPreviewRequest {
+            let txData = jsonData("MultiSendApproveMultihopBatchSwapExactIn")
+            do {
+                let output: T.ResponseType = try self.response(from: txData)
+                completion(.success(output))
+            } catch {
+                completion(.failure(error))
+            }
+        } else {
+            print("---> TransactionPreviewRequest | Unexpected request... -> error")
+            //completion(.failure(InvalidRequestError))
```

Comment:
What is the easiest way to creat a new error class and return it here. For example `InvalidRequestError`?

Reply:
for debug issues, just use a String, I think it is defined as an error protocol implementation somewhere, so you can use it